test/cat.jl

```diff
@@ -118,17 +109,8 @@ module TestCat
     dtc = DataTable(a = NullableArray([2, 3, 4]))
     dtd = DataTable(Any[2:4], [:a])
     dtab = vcat(dta, dtb)
-    dtac = vcat(dta, dtc)
```

Comment:
I'll add this back. I think it failed but I should either convert this to a test for the failure or fix the issue. I think this can be fixed by writing/adding back the `similar_nullable` functions you mentioned elsewhere